src/main.ts

```diff
@@ -450,6 +451,42 @@ export default class ImageWindow extends Plugin {
             })
         );
 
+        this.registerDomEvent(document, "contextmenu", (event: MouseEvent) => {
+            const target = event.target as HTMLElement;
+            if (target.localName !== "img") return;
+
+            const vaultName = this.app.vault.getName().replace(" ", "%20")
```

Comment:
This needs to be either `replaceAll` or `replace(/\s+/g, "%20")` or `encodeURIComponent` or the Obsidian `normalizePath` (probably preferred), although you don't need it with the other change.